pkg/kubelet/kuberuntime/logs/logs.go

```diff
@@ -274,6 +273,51 @@ func (w *logWriter) write(msg *logMessage) error {
 	return nil
 }
 
+// Override tail.FindTailLineStartIndex
+// Dedicated to way too long log that is splitted into multiple lines.
+// In which case, by n logical tail lines we may get more than n lines actually.
+func findLogTailLineStartIndex(f io.ReadSeeker, n int64) (int64, error) {
```

Comment:
Could you add explicit tests for this function please - Otherwise when behavior drifts further from the util package this will become awkward to maintain.

Relatedly is there a reason that we aren't adding a modified version in pkg/util/tail?, this was the only usage of the function in k/k that I can find:

```
[kubernetes(master)] $ rg FindTailLineStartIndex
pkg/kubelet/kuberuntime/logs/logs.go
298:    start, err := tail.FindTailLineStartIndex(f, opts.tail)

pkg/util/tail/tail.go
64:// FindTailLineStartIndex returns the start of last nth line.
69:func FindTailLineStartIndex(f io.ReadSeeker, n int64) (int64, error) {

pkg/util/tail/tail_test.go
44:             s, err := FindTailLineStartIndex(r, test.n)
[kubernetes(master)] $
```

Reply:
I didn't mean to put the modified version here. But I did. There are two reasons. 

First, I think code in util are abstract from specific use case and reusable in many cases. The modified version is dedicated to way too long log. I also discovered that the FindTailLineStartIndex is used once, exactly here. It is really out of my expectation. I think the function is very useful to get the last few lines of a file.

Second, I put the  modified version here because I'm not sure where it should be. It caters to almostly only this situation. I need further advice.

After reviewers' suggestions about the code arrangement, I willl add explicit tests for this function.